lattice-control/lattice-control-interface.smithy

```diff
@@ -0,0 +1,508 @@
+    /// Host ID on which to stop the provider
+    @required    
+    @serialization(name: "host_id")
+    hostId: String,
+
+    /// Reference for the capability provider. Can be any of the forms of 
+    /// uniquely identifying a provider
+    @required    
+    @serialization(name: "provider_ref")
+    providerRef: String,
+
+    /// Link name for this provider
+    @required    
+    @serialization(name: "link_name")
+    linkName: String,
+
+    /// Contract ID of the capability provider
+    @required    
+    @serialization(name: "contract_id")
+    contractId: String,
+
+    /// Optional set of annotations used to describe the nature of this
+    /// stop request
+    annotations: AnnotationMap
+}
+
+/// A command instructing a specific host to perform a live update
+/// on the indicated actor by supplying a new image reference. Note that
+/// live updates are only possible through image references
+structure UpdateActorCommand {
+    /// The host ID of the host to perform the live update
+    @required   
+    @serialization(name: "host_id") 
+    hostId: String,
+
+    /// The actor's 56-character unique ID
+    @required    
+    @serialization(name: "actor_id")
+    actorId: String,
+
+    /// The new image reference of the upgraded version of this actor
+    @required    
+    @serialization(name: "new_actor_ref")
+    newActorRef: String,
+
+    /// Optional set of annotations used to describe the nature of this
+    /// update request. Only actor instances that have matching annotations 
+    /// will be upgraded, allowing for instance isolation by 
+    // autonomous agent deployment spec, for example.
+    annotations: AnnotationMap
+}
+
+/// Standard response for control interface operations
+structure CtlOperationAck {
+    @required
+    accepted: Boolean,
+    @required
+    error: String
+}
+
+/// A list of link definitions
+structure LinkDefinitionList {
+    @required
+    links: ActorLinks
+}
+
+/// A summary representation of a host
+structure Host {
+    @required
+    id: String,
+
+    /// uptime in seconds
+    @required    
+    @serialization(name: "uptime_seconds")
+    uptimeSeconds: U64
+}
+
+/// A response containing the full list of known claims within the lattice
+structure GetClaimsResponse {
+    @required
+    claims: ClaimsList
+}
+
+list ClaimsList {
+    member: ClaimsMap,
+}
+
+map ClaimsMap {
+    key: String,
+    value: String,
+}
+
+/// A request to remove a link definition and detach the relevant actor
+/// from the given provider
+structure RemoveLinkDefinitionRequest {
+     /// The actor's public key. This cannot be an image reference
+    @required    
+    @serialization(name: "actor_id")
+    actorId: String,
+
+    /// The provider's public key. This cannot be an image reference
+    @required    
+    @serialization(name: "provider_id")
+    providerId: String,
+
+    /// The provider's link name
+    @required    
+    @serialization(name: "link_name")
+    linkName: String,
+}
```

Comment:
Just a note, this is going to require a breaking change to the control interface client as a remove/delete request takes the actor ID, contract ID, and link name. https://docs.rs/wasmcloud-control-interface/0.4.4/wasmcloud_control_interface/struct.Client.html#method.remove_link

I think this structure is actually a better representation of the operation, just noting that's how it's done today. I believe this is due to how we store the link definition data in OTP (the actor ID, contract ID, and link name are a composite key in a map).

Reply:
yeah, everything basically downwind/downhill from this new crate will end up taking a full semver bump because of the compatibility breaks, including things like annotations.